src/metadataservice_go/internal/keyvaluestore/utilities.go

```diff
@@ -0,0 +1,10 @@
+package keyvaluestore
+
+import (
+	"github.com/IBM/gedsmds/protos"
+	"strings"
+)
+
+func (kv *Service) getNestedPath(objectId *protos.ObjectID) []string {
+	return strings.Split(objectId.GetKey(), commonDelimiter)
+}
```

Comment:
I would avoid splitting the string into common prefixes. This removes the possibility to do a prefix search in leveldb.